tests/test_common.py

```diff
@@ -306,3 +306,28 @@ def poll_callback(request):
 
     list(client.poll(collection_name="X", uri="/poll"))
     assert client.jwt_token == second_token
+
+
+@pytest.mark.parametrize("version", [11, 10])
+@responses.activate
+def test_jwt_token_when_set_directly(version):
+    given_token = "abcd"
+    client = make_client(version)
+
+    # The purpose of this test is to check that this assignment has effect:
+    client.jwt_token = given_token
+
+    def poll_callback(request):
+        _, _, token = request.headers["Authorization"].partition("Bearer ")
+        assert token == given_token
+        return (200, make_taxii_headers(version), get_fix(version).POLL_RESPONSE)
+
+    responses.mock._matches.append(
+        responses.CallbackResponse(
+            responses.POST,
+            url="http://example.localhost/poll",
+            callback=poll_callback,
+            stream=True,
+        )
+    )
+    list(client.poll(collection_name="X", uri="/poll"))
```

Comment:
is this intended to be like this? or did you want to `assert` something?

or is it a generator that you want to exhaust? in that case maybe

```
for _ in client.poll(...):
    pass
```

but it's not that important

Reply:
Assert is in the callback which is called when the stream from `poll` is consumed.